services/ec2/src/main/java/org/finra/gatekeeper/services/accessrequest/AccessRequestService.java

```diff
@@ -294,6 +307,64 @@ When the request was actioned by an approver (or canceled by a user/approver)
         return (List<CompletedAccessRequestWrapper>)filterResults(results);
     }
 
+
+    /**
+     *
+     * @param users         List of User IDs to return live requests for
+     * @param eventType     Type of event that invokes this method. Allowed values are: EventType.APPROVAL and EventType.EXPIRATION
+     * @param expiredRequest If a request is expiring, this parameter contains the request information for the expired request.
+     * @return              A list of all live and recently expired access requests.
+     */
+    public List<ActiveRequestUser> getLiveRequests(List<User> users, EventType eventType, AccessRequest expiredRequest) {
+        logger.info("Fetching live requests.");
+
+    /*  This object is all of the Activiti Variables associated with the request
+        This map will contain the following:
+        When the request was opened
+        When the request was actioned by an approver (or canceled by a user/approver)
+        How many attempts it took to approve the user*/
+        Map<String, Map<String, Object>> historicData = new HashMap<>();
+
+        //we have to map the activiti hitory items to the gatekeeper requests map activiti objects to the access requests
+        Map<String, Long> activitiAccessRequestMap = new HashMap<>();
+        //map gatekeeper access request ids
+        Map<Long, AccessRequest> gkAccessRequestMap = new HashMap<>();
+
+    /*this gets all of the access requests history items. We exclude variable initialization
+        because activiti will break if we ever change the access request object so we'll just fetch the items
+        ourselves */
+        List<HistoricVariableInstance> accessRequests = getAllHistoricAccessRequests();
```

Comment:
getLiveRequests should only query access requests from the maximum request time, this is pulling the whole dataset, it will be slow if we do it like this.